### PR TITLE
Expose the ResourceConsumptionAttribute type as a public type

### PR DESCRIPTION
**Pull request by AlexGhiondea**

### CODE REVIEW

**Comment by AlexGhiondea** [comment]
@weshaggard could you take a quick look at this? Changing this into an ApiRoot makes the type public after bclrewriter.

**Comment by AlexGhiondea** [comment]
@dotnet-bot please test Linux ARM Emulator Cross Release Build (machine issue)

File: src/mscorlib/model.xml [hunk 1]
```diff
@@ -10448,7 +10448,7 @@
       <Member Name="RemoveEventHandler&lt;T&gt;(System.Action&lt;System.Runtime.InteropServices.WindowsRuntime.EventRegistrationToken&gt;,T)" />
       <Member Name="StringToHString(System.String)" />
     </Type>
-    <Type Status="ImplRoot" Name="System.Runtime.Versioning.ResourceConsumptionAttribute">
+    <Type Status="ApiRoot" Name="System.Runtime.Versioning.ResourceConsumptionAttribute">
```
Comment (weshaggard) [inline src/mscorlib/model.xml]: You can remove this if the plan is to make it public.

**Comment by weshaggard** [comment]
Is there a better home for this instead of S.P.CoreLib?

**Comment by AlexGhiondea** [comment]
@weshaggard it looks like that, and the other types inside System.Runtime.Versioning can be pulled out of S.P.corelib. I will go with that plan instead.

**Comment by jkotas** [comment]
> it looks like that, and the other types inside System.Runtime.Versioning can be pulled out of S.P.corelib

Agree.

**Pull request closed without merging**